src/com/goide/inspections/GoTestingDeclarationsInspection.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2013-2015 Sergey Ignatov, Alexander Zolotov, Mihai Toader, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.GoConstants;
+import com.goide.psi.*;
+import com.goide.psi.impl.GoElementFactory;
+import com.goide.runconfig.testing.GoTestFinder;
+import com.intellij.codeInspection.LocalQuickFixBase;
+import com.intellij.codeInspection.ProblemDescriptor;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.PsiElement;
+import com.intellij.util.containers.ContainerUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class GoTestingDeclarationsInspection extends GoInspectionBase {
+  private enum TestingType {TEST, TEST_MAIN, BENCHMARK, EXAMPLE}
+
+  private static TestingType getTestingType(@NotNull String name) {
+    if (name.startsWith(GoConstants.TEST_PREFIX)) {
+      return name.toLowerCase().equals("testmain") ? TestingType.TEST_MAIN : TestingType.TEST;
+    }
+    if (name.startsWith(GoConstants.BENCHMARK_PREFIX)) return TestingType.BENCHMARK;
+    if (name.startsWith(GoConstants.EXAMPLE_PREFIX)) return TestingType.EXAMPLE;
+    return null;
+  }
+
+  private static String getPrefixFromType(@NotNull TestingType type) {
+    switch (type) {
+      case TEST:
+      case TEST_MAIN:
+        return GoConstants.TEST_PREFIX;
+      case BENCHMARK:
+        return GoConstants.BENCHMARK_PREFIX;
+      case EXAMPLE:
+      default:
+        return GoConstants.EXAMPLE_PREFIX;
+    }
+  }
+
+  @Override
+  protected void checkFile(@NotNull final GoFile file, @NotNull ProblemsHolder problemsHolder) {
+    if (!GoTestFinder.isTestFile(file)) return;
+    GoImportSpec testingImportSpec = file.getImportedPackagesMap().get(GoConstants.TESTING_PATH);
+    String testingAlias = null;
+    if (testingImportSpec != null) {
+      testingAlias = testingImportSpec.getAlias();
+      testingAlias = testingAlias == null ? testingImportSpec.getPath() : testingAlias;
```

Comment:
getLocalPackageName is more appropriate than getPath.
+ should be tested with dot-imports line `import . "testing"`